tests/utils/utils.go

```diff
@@ -323,6 +321,80 @@ func CreateReceiveCrossChainMessageTransaction(
 	return SignTransaction(destinationTx, senderKey, subnetInfo.EVMChainID)
 }
 
+// Constructs a transaction to call addProtocolVersion
+// Returns the signed transaction.
+func CreateAddProtocolVersionTransaction(
+	ctx context.Context,
+	signedMessage *avalancheWarp.Message,
+	teleporterRegistryAddress common.Address,
+	senderKey *ecdsa.PrivateKey,
+	subnetInfo interfaces.SubnetTestInfo,
+) *types.Transaction {
+	// Construct the transaction to send the Warp message to the destination chain
+	log.Info("Constructing addProtocolVersion transaction for the destination chain")
+	// numSigners, err := signedMessage.Signature.NumSigners()
+	// Expect(err).Should(BeNil())
+
+	callData, err := teleporterregistry.PackAddProtocolVersion(0)
+	Expect(err).Should(BeNil())
+
+	gasFeeCap, gasTipCap, nonce := CalculateTxParams(ctx, subnetInfo, PrivateKeyToAddress(senderKey))
+
+	destinationTx := predicateutils.NewPredicateTx(
+		subnetInfo.EVMChainID,
+		nonce,
+		&teleporterRegistryAddress,
+		500_000,
```

Comment:
How'd you arrive at 500_000? Curious because I'm implementing similar logic in the relayer to handle off chain messages.

Reply:
Initially had the e2e test with too little gas, and in the error message it told me the expected gas